src/components/RefinementList/RefinementList.tsx

```diff
@@ -313,7 +322,7 @@ class RefinementList<TTemplates extends Templates> extends Component<
           placeholder={this.props.searchPlaceholder}
           disabled={shouldDisableSearchBox}
           cssClasses={this.props.cssClasses.searchable!}
-          templates={templates}
+          templates={{ ...defaultSearchBoxTemplates, ...templates }}
```

Comment:
That would avoid having line 316?

```suggestion
          templates={{ ...defaultSearchBoxTemplates, ...this.props.searchBoxTemplateProps.templates }}
```